File: Backend/src/main/java/com/luckyseven/backend/sharedkernel/jwt/repository/BlackListTokenRepository.java
package com.luckyseven.backend.sharedkernel.jwt.repository;

import com.luckyseven.backend.sharedkernel.jwt.entity.BlackListToken;
import jakarta.transaction.Transactional;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface BlackListTokenRepository extends JpaRepository<BlackListToken,Long> {

 Boolean existsByTokenValue(String tokenValue);

 Optional<BlackListToken> findByTokenValue(String tokenValue);

}

File: Backend/src/main/kotlin/com/luckyseven/backend/sharedkernel/jwt/repository/BlackListTokenRepository.kt
package com.luckyseven.backend.sharedkernel.jwt.repository

import com.luckyseven.backend.sharedkernel.jwt.entity.BlackListToken
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface BlackListTokenRepository : JpaRepository<BlackListToken, Long> {
    fun findByTokenValue(tokenValue: String): BlackListToken?
} 